src/test/java/org/jabref/logic/texparser/TexParserTest.java

```diff
@@ -0,0 +1,294 @@
+        TexParserResult expectedResult = new TexParserResult();
+
+        if (match) {
+            expectedResult.getCitations().put(key, new ArrayList<>());
+            expectedResult.getCitations().get(key).add(
+                    new Citation(Paths.get("foo/bar"), 1, 0, citeString.length(), citeString));
+        }
+
+        assertEquals(expectedResult, texParserResult);
+    }
+
+    private void testMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, true);
+    }
+
+    private void testNonMatchCite(String citeString) {
+        testCite(UNRESOLVED, citeString, false);
+    }
+
+    @Test
+    void testCiteCommands() {
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\cite*{UnresolvedKey}");
+        testMatchCite("\\parencite[post]{UnresolvedKey}");
+        testMatchCite("\\cite[pre][post]{UnresolvedKey}");
+        testMatchCite("\\citep{UnresolvedKey}");
+
+        testNonMatchCite("\\citet21312{123U123n123resolvedKey}");
+        testNonMatchCite("\\1cite[pr234e][post]{UnresolvedKey}");
+        testNonMatchCite("\\citep55{5}UnresolvedKey}");
+        testNonMatchCite("\\cit2et{UnresolvedKey}");
+    }
+
+    private void addCite(CrossingKeysResult expectedResult, String key, Path texFile, int line, int colStart, int colEnd, String lineText, boolean insert) {
+        Citation citation = new Citation(texFile, line, colStart, colEnd, lineText);
+
+        if (!expectedResult.getParserResult().getCitations().containsKey(key)) {
+            expectedResult.getParserResult().getCitations().put(key, new ArrayList<>());
+        }
+
+        if (!expectedResult.getParserResult().getCitations().get(key).contains(citation)) {
+            expectedResult.getParserResult().getCitations().get(key).add(citation);
+        }
+
+        if (insert) {
+            BibEntry clonedEntry = (BibEntry) expectedResult.getMasterDatabase().getEntryByKey(key).get().clone();
+            expectedResult.getNewDatabase().insertEntry(clonedEntry);
+        } else {
+            expectedResult.getUnresolvedKeys().add(key);
+        }
+    }
+
+    @Test
+    void testSingleFile() throws URISyntaxException, IOException {
+        InputStream originalStream = TexParserTest.class.getResourceAsStream("origin.bib");
+        Path texFile = Paths.get(TexParserTest.class.getResource("paper.tex").toURI());
+
+        try (InputStreamReader originalReader = new InputStreamReader(originalStream, StandardCharsets.UTF_8)) {
+            ParserResult result = new BibtexParser(importFormatPreferences, new DummyFileUpdateMonitor()).parse(originalReader);
+
+            TexParserResult parserResult = new DefaultTexParser().parse(texFile);
+            TexParserResult expectedParserResult = new TexParserResult();
+            expectedParserResult.getFileList().add(texFile);
+
+            CrossingKeysResult crossingResult = new CrossingKeys(parserResult, result.getDatabase()).resolveKeys();
```

Comment:
Tests for the crossreferencing should be in another class (CrossingKeysTest). With unit tests, you want to test each element in isolation, i.e. parser extracts correct citation and crossing keys assigns correct entries.
It is also good to have a few more global tests that make sure the whole process works (these are called integration tests), but these should be in a separate testclass.